src/MICore/MIResults.cs

```diff
@@ -1203,9 +1209,9 @@ private ResultListValue ParseResultList(Span input, out Span rest)
 
         private void ParseError(string message, Span input)
         {
```

Comment:
knowing this method exists, can it be reused for the above? It adds more logging in this case. 

I think getting rid of the `#if DEBUG` block might be good too as this error would be extremely useful in a user's bug case.